src/arrayMethodSort.js

```diff
@@ -4,8 +4,22 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  const defaultCompare = function defaultCompare(a, b) {
+    return a.toString() > b.toString();
+  };
+
+  [].__proto__.sort2 = function (compareFunction = defaultCompare) {
+    const compare = compareFunction;
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (compare(this[j], this[j + 1]) > 0) {
+          [this[j], this[j + 1]] = [this[j + 1], this[j]];
+        }
+      }
+    }
+
+    return this;
```

Comment:
While a bubble sort is a valid approach and meets the task requirements, it is not the most efficient sorting algorithm for larger arrays. If performance is a concern, consider researching and implementing a more efficient algorithm, such as quicksort or mergesort, in the future.